include/node.tcc

```diff
@@ -547,11 +547,13 @@ void mpm::Node<Tdim, Tdof,
   // iterate over all materials in the material_ids set and update the change in
   // momentum
   std::lock_guard<std::mutex> guard(node_mutex_);
-  for (auto mat_itr = material_ids_.begin(); mat_itr != material_ids_.end();
-       ++mat_itr)
-    property_handle_->update_property(
-        "change_in_momenta", prop_id_, *mat_itr,
-        velocity_ * property_handle_->property("masses", prop_id_, *mat_itr) -
-            property_handle_->property("momenta", prop_id_, *mat_itr, Tdim),
-        Tdim);
+  for (auto mitr = material_ids_.begin(); mitr != material_ids_.end(); ++mitr) {
+    Eigen::MatrixXd mass =
```

Comment:
could you make these const if possible use the size: `Eigen::Matrix<double, Tdim, 1>` or  `Eigen::Matrix<double, 1, 1>`